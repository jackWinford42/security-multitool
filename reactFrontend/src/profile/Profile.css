/* profile page styling */

#profile_page_pic {
    border-radius: 50%;
}

#profile_content {
    padding: 2em;
    margin-top: 3em;
}

#right_profile_div {
    font-size: 1.5em;
    text-align: center;
}

.form_element {
    border-bottom: 4px black groove;
    text-align: center;
    margin: 0 auto;
    font-size: 1.5em;
    margin-top: 1.5em;
}

.form_element span {
    font-weight: bold;
    text-align: center;
}

.skinny {
    margin: 0 auto;
    width: auto;
}

.short_input {
    width: 40%;
    margin: 0 auto;
}
/* end profile page styling */

/* begin UserHistory styling */

.UserHistory ul {
    width: 700px;
    margin: 0 auto;
}