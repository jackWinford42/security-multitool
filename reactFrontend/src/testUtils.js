import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };