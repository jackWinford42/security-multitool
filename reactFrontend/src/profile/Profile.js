import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import RamtApi from "../common/Api";
import UserHistory from "./UserHistory";
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import "./Profile.css"

export default function Profile({logout}) {
  const user = useSelector(st => st.currUser);
  const history = useHistory();
  const [isLoading, setIsLoading] = useState(false);
  const [empty, setEmpty] = useState(true);
  const [displayHistory, setDisplayHistory] = useState({});
  const [open, setOpen] = useState(false);

  useEffect(() => {
    async function getHistory() {
      try {
        const userHistory = await RamtApi.getUserHistory(user.email);
        setDisplayHistory(userHistory)
        setEmpty(false);
      } catch (err) {
        console.error("UserHistory get: issue loading history", err);
      }
    }
    setIsLoading(true);
    if (empty) getHistory()
    setIsLoading(false);
  }, [empty, user])

  const goToEdit = () => {
    let path = `/edit-profile`; 
    history.push(path);
  }

  const deleteUser = () => {
    RamtApi.delCurrUser(user.email)
    history.push("/");
    logout();
  }

  const handleOpen = () => {
    setOpen(true)
  };

  const handleClose = () => {
    setOpen(false)
  };

  const dumpHistory = () => {
    RamtApi.dumpUserHistory(user.email);
    setEmpty(true);
  }

  if (isLoading) return <p>Loading &hellip;</p>;
  return (
    <div className="Profile">
      <div id="title">
        <h1>Profile Page</h1>
      </div>
      <div id="profile_content">
        <div className="row">
          <div className="col-4">
            <img id="profile_page_pic" src={ user.profile_pic } width="250" height="250" alt={ user.username }/>
          </div>
          <div className="col-2"></div>
          <div className="col-5" id="right_profile_div">
            <ul className="list-group list-group-flush">
              <li className="list-group-item"><span>Display name: { user.username }</span></li>
              <li className="list-group-item"><span>Email: { user.email }</span></li>
              <li className="list-group-item">
                <button onClick={goToEdit} className="btn btn-lrg btn-warning">Edit</button>
              </li>
              <li className="list-group-item">
                <button onClick={handleOpen} className="btn btn-lrg btn-danger">Delete Account</button>
              </li>
              <li className="list-group-item">
                <button onClick={dumpHistory} className="btn btn-lrg btn-danger">Clear Your History</button>
              </li>
            </ul>
          </div>
          <div className="col-1"></div>
        </div>
      </div>
      { displayHistory.history && <UserHistory HistoryItems={displayHistory.history.rows}/>}
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="dialog-title"
        aria-describedby="dialog-description"
      >
        <DialogTitle id="dialog-title">
          Are you sure you would like to delete your account?
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="dialog-description">
            Deleting your account is permanent and will remove your history.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} autoFocus>
            no
          </Button>
          <Button onClick={deleteUser} autoFocus>
            yes
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}