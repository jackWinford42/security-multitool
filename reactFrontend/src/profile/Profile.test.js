import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Profile/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Profile/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});