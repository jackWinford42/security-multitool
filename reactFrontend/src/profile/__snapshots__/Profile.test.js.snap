// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <div
      id="title"
    >
      <h1>
        Profile Page
      </h1>
    </div>
    <div
      id="profile_content"
    >
      <div
        class="row"
      >
        <div
          class="col-4"
        >
          <img
            height="250"
            id="profile_page_pic"
            width="250"
          />
        </div>
        <div
          class="col-2"
        />
        <div
          class="col-5"
          id="right_profile_div"
        >
          <ul
            class="list-group list-group-flush"
          >
            <li
              class="list-group-item"
            >
              <span>
                Display name: 
              </span>
            </li>
            <li
              class="list-group-item"
            >
              <span>
                Email: 
              </span>
            </li>
            <li
              class="list-group-item"
            >
              <button
                class="btn btn-lrg btn-warning"
              >
                Edit
              </button>
            </li>
            <li
              class="list-group-item"
            >
              <button
                class="btn btn-lrg btn-danger"
              >
                Delete Account
              </button>
            </li>
            <li
              class="list-group-item"
            >
              <button
                class="btn btn-lrg btn-danger"
              >
                Clear Your History
              </button>
            </li>
          </ul>
        </div>
        <div
          class="col-1"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
